Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1899,22 +1905,21 @@ private InvalidBlockException ValidateNextBlock(Block<T> nextBlock)
             if (nextBlock.Index != index)
             {
                 return new InvalidBlockIndexException(
-                    $"The expected block index is #{index}, but its index " +
-                    $"is #{nextBlock.Index}.");
+                    $"The expected index of block [{nextBlock.Hash}] is #{index}, " +
+                    $"but its index is #{nextBlock.Index}.");
             }
 
             if (nextBlock.Difficulty < difficulty)
             {
                 return new InvalidBlockDifficultyException(
-                    $"The expected difficulty of the block #{index} " +
-                    $"is {difficulty}, but its difficulty is " +
-                    $"{nextBlock.Difficulty}.");
+                    $"The expected difficulty of the block #{index} [{nextBlock.Hash}] " +
```

Comment:
It would be better if `InvalidBlockDifficultyException` has more properties like `BlockHash` and `BlockIndex` in the future…

```suggestion
                    $"The expected difficulty of the block #{index} {nextBlock.Hash} " +
```

Reply:
How about make `InvalidBlockException` to have `BlockHash` and `BlockIndex`?

Reply:
> How about make `InvalidBlockException` to have `BlockHash` and `BlockIndex`?

That sounds good if it's possible. Some of its subtypes might be unable to have these properties though?